packages/widgets/src/blocks/legacy-widget/edit/index.js

```diff
@@ -136,10 +136,11 @@ function NotEmpty( {
 
 	return (
 		<>
-			{ idBase === 'text' && (
+			{ idBase && instance?.raw && (
```

Comment:
I just checked and third party widgets don't seem to haw "raw". How curious! A comment explaining why that's the case might be helpful. Or even better, maybe we would even check idBase against the list of available conversions – that would be obvious on the first glance.

Reply:
Third party widgets have to set `show_instance_in_rest = true`. It's in the block editor handbook somewhere. I've added a comment in 9b241ad323015a18be07f4036d2ae6d91ea6b005 too 🙂

Reply:
Nice, thank you for adding that comment! But I have one more concern :D 

If a third party widgets sets `show_instance_in_rest = true`, the "Convert to blocks" button will be displayed, even though there is no transform to handle it, right?

Reply:
Mm! Very good flag. This approach doesn't work at all with third party transforms 😬

I see three options:

1. We could hide the _Convert_ button for third party widgets. This seems too limiting, though.
2. We could have the _Convert_ button invoke the first transform it finds. In this way the _Convert to blocks_ button is essentially a more visible alternative to the transforms menu which can also transform text widgets. 
3. We could ditch the _Convert_ button and have only transforms. See https://github.com/WordPress/gutenberg/pull/34610.

If I can figure out a way to make the transforms menu say "Blocks" instead of "Paragraph" (see https://github.com/WordPress/gutenberg/pull/34610#issuecomment-914488926) then I like the idea of (3).

Reply:
@noisysocks I like (3) because `convert` API looks almost exactly the same as `transforms` API. At the same time, I like having the _Convert_ button because it makes it apparent that I can migrate and even encourages me a bit. It is not hidden behind a hamburger menu.

Could we keep the Convert button as it is for widgets from the list, and make it do (2) for third party widgets?